Fix issues in release 3.0

hoyosjs opened the pull request:
Changes discussed to obtain green CI and fix RHEL6 outerloop builds:

- Restore the Helix SDK using Tools.props to work around msbuild issues
- Disable formatting jobs
- Disable windows ARM jobs until dotnet/runtime#1097 is solved
- Fix TRACE statement for RHEL6 in debug/checked pal out-of-proc unwind.

cc @jeffschwMSFT @mmitche @jashook @trylek

hoyosjs committed:
Restore Helix SDK using Tools.props to work around MSBuild race condition

hoyosjs committed:
Disable formatting jobs

hoyosjs committed:
Disable ARM windows jobs

hoyosjs committed:
Fix tracing in RHEL6 build

hoyosjs commented:
/azp run coreclr-outerloop

trylek reviewed (approved):
LGTM, thanks so much for keeping us in reasonable shape for servicing releases!

hoyosjs committed:
Disable additional Win-ARM test run in CI

Change `platformGroup: all` in CI to explicit platform list and disable win-arm

hoyosjs commented:
/azp run coreclr-outerloop

jamshedd commented:
Approved for Feb.
